Expensify/App
# Fix Pusher Re-Authentication bug

### Details
Well explained in the linked issue.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/152022

### Tests
**Testing subscribing with bad session**
1. Invalidate your authToken by messing with it in `localStorage`
1. Filter in the JS console logs for `Pusher`
1. Tap on a chat in the LHN
1. Verify that your logs look like this and that Pusher has successfully resubscribed to all channels
<img width="690" alt="Screen Shot 2021-01-22 at 1 47 54 PM" src="https://user-images.githubusercontent.com/32969087/105560720-b841f400-5cb8-11eb-92d0-b453d3a6722d.png">

**Testing loss of session and connectivity** 
1. Invalidate your authToken by messing with it in `localStorage`
2. Turn off your WiFi
3. Filter in the JS console logs for `Pusher`
4. Wait until you see this log line in the JS console
<img width="692" alt="Screen Shot 2021-01-22 at 1 52 17 PM" src="https://user-images.githubusercontent.com/32969087/105560855-2c7c9780-5cb9-11eb-9a8d-bf2f5d294588.png">
5. Turn WiFi back on
6. Don't touch anything! Wait and then verify that you are eventually resubscribed to Pusher
<img width="676" alt="Screen Shot 2021-01-22 at 1 52 49 PM" src="https://user-images.githubusercontent.com/32969087/105560884-474f0c00-5cb9-11eb-92ea-d6fb40af427c.png">

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
